src/header-navigation/examples.js

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+/* eslint-disable react/display-name*/
+
+import * as React from 'react';
+import {withProps} from '../helpers';
+import {Button} from '../button';
+import {StatefulMenu as Menu, KEY_STRINGS} from '../menu';
+import {StatefulSelect as Search} from '../select';
+import {styled} from '../styles';
+import COLORS from '../select/examples-colors';
+import {HamburgerIcon} from './examples-icons';
+
+import {
+  HeaderNavigation,
+  ALIGN,
+  StyledNavigationItem as NavigationItem,
+  StyledNavigationList as NavigationList,
+} from './index';
+import tests from './examples-list';
+import {TYPE} from '../select';
+
+const Link = withProps(
+  styled('a', props => ({
+    ':hover': {color: props.$theme.colors.primary400},
+  })),
+  {tabIndex: '0'},
+);
+// $FlowFixMe
+Link.displayName = 'Link';
+
+const Hamburger = styled('div', props => ({
+  lineHeight: 'initial',
+}));
+Hamburger.displayName = 'Hamburger';
+
+let ITEMS = [];
+for (let i = 0; i < 100; i++) {
+  ITEMS.push({label: `Item ${i}`});
+}
+const options = {
+  options: COLORS,
+  labelKey: 'id',
+  valueKey: 'color',
+  placeholder: 'Choose a color',
+  maxDropdownHeight: '300px',
+};
+class MenuContainer extends React.Component<
+  {isOpen: boolean, align: string},
+  {isOpen: boolean},
+> {
+  static defaultProps = {
+    isOpen: false,
+    align: '',
+  };
+  state = {
+    isOpen: this.props.isOpen,
+  };
+  toggleMenu() {
+    this.setState({
+      isOpen: !this.state.isOpen,
+    });
+  }
+  render() {
+    return (
+      <React.Fragment>
+        <Hamburger
+          tabIndex="0"
+          aria-label="Navigation menu"
+          onKeyDown={e => {
+            switch (e.key) {
+              case KEY_STRINGS.ArrowDown:
+              case KEY_STRINGS.Space:
+              case KEY_STRINGS.Enter:
+                this.toggleMenu();
+                return;
+            }
+          }}
+          onClick={() => this.toggleMenu()}
+        >
+          {HamburgerIcon()}
+        </Hamburger>
+        {this.state.isOpen && (
+          <Menu
+            items={ITEMS}
+            overrides={{
+              List: {
+                style: {
+                  cursor: 'pointer',
+                  position: 'absolute',
+                  height: '90vh',
+                  right: this.props.align === ALIGN.right ? '0' : 'initial',
+                },
+              },
+            }}
+          />
+        )}
+      </React.Fragment>
+    );
+  }
+}
+
+export default {
+  [tests.HEADER_NAVIGATION_WITH_MENU_ELEMENT]: () => {
+    return (
+      <React.Fragment>
+        <div style={{marginTop: '50px'}} />
+        <HeaderNavigation>
+          <NavigationList $align={ALIGN.left}>
```

Comment:
Sorry, I must've missed this one on a previous look through. Why is this prop `$` prefixed? It seems like there should be a container that would do the `$` mapping

Reply:
this is a styled component. If we add container, it would add complexity to maintain things.

Reply:
I'm pretty torn on this. It could be confusing to users if some of the primary exports for a component require `$` prefixes. On the other hand I agree creating wrappers every time we use this pattern could be annoying.

A couple ideas:
1. Write an HOC that wraps styled components and maps the $ props:
```js
import {asPrimaryExport} from '../styled/as-primary-export-hoc';

export const StyledNavigationList = styled('div', {...});

export const NavigationList = asPrimaryExport(StyledNavigationList, ['align', 'prop2', 'prop3']);
```
This would be far less tedious than maintaining a wrapper component for every time we use this pattern.

2. Add logic to our `styled` implementation that supports similar prop mapping
```js
export const StyledNavigationList = styled('div', {...}, {
  mapProps: ['align', 'prop2', 'prop3']
});
```
Internally I think this could use styletron's [withWrapper](https://github.com/styletron/styletron/tree/master/packages/styletron-react#withwrapper)? Still not 100% sure. Also adds more complexity to our styled implementation.

--

Obviously those are outside the scope of this specific PR, just trying to think of the best long term solution. Maybe short term we bite the bullet and build a container manually?

Reply:
apparently, my last comment wasn't submitted when I replied

Reply:
I applied 1) approach. Currently need to fix the test using it